pkg/util/merger.go

```diff
@@ -2,6 +2,29 @@ package util
 
 import "github.com/prometheus/common/model"
 
+// MergeNSamples merges and dedupes n sets of already sorted sample pairs.
+func MergeNSamples(sampleSets ...[]model.SamplePair) []model.SamplePair {
+	l := len(sampleSets)
+	switch l {
+	case 0:
```

Comment:
Unless I've missed it, and even though that's trivial, I don't think there is an unit test covering this `case` statement.
If I understood well, we'd need this additional test case:
```
		{
			sampleSets: [][]model.SamplePair{},
			expected:   []model.SamplePair{},
		},
```